core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1636,6 +1637,10 @@ private void restoreFile(final BlobStoreIndexShardSnapshot.FileInfo fileInfo, fi
                 } else {
                     stream = new RateLimitingInputStream(partSliceStream, restoreRateLimiter, restoreRateLimitingTimeInNanos::inc);
                 }
+
```

Comment:
I *think*, this can happen if you fail a shard and it's not cleaned up yet, the restore process tries to _reuse_ files which is mostly bogus. I think higher up we should call `Lucene#pruneUnreferencedFiles` but I think @imotov should jump in here. Let's open a new issue for this ok? this must not block this upgrade